src/gdb.ts

```diff
@@ -68,11 +64,47 @@ class GDBDebugSession extends MI2DebugSession {
 			this.isSSH = true;
 			this.trimCWD = args.cwd.replace(/\\/g, "/");
 			this.switchCWD = args.ssh.cwd;
-			this.miDebugger.ssh(args.ssh, args.ssh.cwd, args.target, args.arguments, args.terminal, false).then(() => {
-				if (args.autorun)
-					args.autorun.forEach(command => {
-						this.miDebugger.sendUserInput(command);
-					});
+		}
+		if (args.autorun === undefined)
+			args.autorun = [];
+
+		if (!hasAutorunBeforeArgs)
+			args.autorunBefore = ["gdb-set target-async on",
+				"environment-directory \"$cwd\""];
+
+		if (attach) {
+			this.attached = !args.remote;
+			if (!hasAutorunBeforeArgs)
+				args.autorunBefore.push("target-select remote $target");
+
+		} else {
+			this.attached = false;
+			this.needContinue = true;
+			if (!hasAutorunBeforeArgs)
+				args.autorunBefore.push("target-select $target");
+		}
+		args.autorun = args.autorun.map(
+			s => {return escape(s);
+		});
+
+		args.autorunBefore = args.autorunBefore.map(
```

Comment:
I have a set of commands to send to gdb after connection
"load ${workspaceRoot}\\bin\\binary.x"
if i don't escape the path is wrong it remove all the \
the final command is
"load c:somedirAsomedirB\bin\binary.x" and not c:\somedirA\somedirB\bin\binary.x